React/my-app/src/screens/sign_up/index.js

```diff
@@ -106,12 +107,18 @@ class SignUpContainer extends Component {
           last_name: surname,
           locale: "en"
         }
-      },
-      this.onSuccess,
-      this.onFailure
+      })
     );
   }
 
+  componentWillReceiveProps(newProps) {
+    if (newProps.error) {
```

Comment:
i'd replace it by 
`this.setState(newProps.error ? { emailError: MAIL_IN_USE } : { redirect: true })`